staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -439,33 +452,79 @@ func (p *patcher) applyPatch(_ context.Context, _, currentObject runtime.Object)
+	p.trace.Step("About to check admission control")
+	if p.mutatingAdmission != nil && p.mutatingAdmission.Handles(admission.Update) {
+		userInfo, _ := request.UserFrom(ctx)
+		return updatedObject, p.mutatingAdmission.Admit(admission.NewAttributesRecord(updatedObject, currentObject, p.kind, p.namespace, p.name, p.resource, p.subresource, admission.Update, userInfo))
+	}
+	return updatedObject, nil
+}
+
+type patchCreater struct {
+	*patcher
+}
+
+func (p *patchCreater) executeOperation(ctx context.Context) (runtime.Object, error) {
+	versionedObjToCreate, err := p.creater.New(p.kind)
+	if err != nil {
+		return nil, err
+	}
+	if err := runtime.DecodeInto(p.codec, p.patchBytes, versionedObjToCreate); err != nil {
+		return nil, err
+	}
+	unversionedObjToCreate, err := p.toUnversioned(versionedObjToCreate)
+	if err != nil {
+		return nil, err
+	}
+	if err := checkName(unversionedObjToCreate, p.name, p.namespace, p.namer); err != nil {
+		return nil, err
+	}
+	if _, err := p.applyAdmission(ctx, unversionedObjToCreate, nil); err != nil {
+		return nil, err
+	}
+	return p.restApplier.Create(ctx, unversionedObjToCreate, p.createValidation, false)
+}
+
+func (p *patchCreater) applyAdmission(ctx context.Context, objToCreate runtime.Object, _ runtime.Object) (runtime.Object, error) {
 	p.trace.Step("About to check admission control")
-	return patchedObject, p.admissionCheck(patchedObject, currentObject)
+	if p.mutatingAdmission != nil && p.mutatingAdmission.Handles(admission.Create) {
+		userInfo, _ := request.UserFrom(ctx)
+		return objToCreate, p.mutatingAdmission.Admit(admission.NewAttributesRecord(objToCreate, nil, p.kind, p.namespace, p.name, p.resource, p.subresource, admission.Create, userInfo))
+	}
+	return objToCreate, nil
 }
 
 // patchResource divides PatchResource for easier unit testing
 func (p *patcher) patchResource(ctx context.Context, scope RequestScope) (runtime.Object, error) {
 	p.namespace = request.NamespaceValue(ctx)
 	switch p.patchType {
 	case types.JSONPatchType, types.MergePatchType:
-		p.mechanism = &jsonPatcher{patcher: p}
+		mechanism := &jsonPatcher{patcher: p}
+		p.operation = &patchUpdater{patcher: p, mechanism: mechanism}
 	case types.StrategicMergePatchType:
 		schemaReferenceObj, err := p.unsafeConvertor.ConvertToVersion(p.restPatcher.New(), p.kind.GroupVersion())
 		if err != nil {
 			return nil, err
 		}
-		p.mechanism = &smpPatcher{patcher: p, schemaReferenceObj: schemaReferenceObj}
+		mechanism := &smpPatcher{patcher: p, schemaReferenceObj: schemaReferenceObj}
+		p.operation = &patchUpdater{patcher: p, mechanism: mechanism}
 	// this case is unreachable if ServerSideApply is not enabled because we will have already rejected the content type
 	case types.ApplyPatchType:
-		p.mechanism = &applyPatcher{patcher: p, model: scope.OpenAPISchema}
+		_, err := p.restApplier.Get(ctx, p.name, &metav1.GetOptions{})
+		if errors.IsNotFound(err) && p.restApplier != nil {
```

Comment:
if `p.restApplier` is nil, you'll never reach this line (because of the line before). Also, if `restApplier` is nil, would it mean you want to use updated to create?

Reply:
Oh, right, then it'll use the updater and fail right away, so that's fine. You still need to fix the crash :-)

Reply:
This is a bug, the previous line should be using the restPatcher

Reply:
good catch